azurerm/internal/services/apimanagement/api_management_identity_provider_aadb2c_resource_test.go

```diff
@@ -0,0 +1,191 @@
+	data := acceptance.BuildTestData(t, "azurerm_api_management_identity_provider_aadb2c", "test")
+	r := ApiManagementIdentityProviderAADB2CResource{}
+	b2cConfig := testAccAzureRMApiManagementIdentityProviderAADB2C_getB2CConfig(t)
+	apiDomain := acceptance.AzureProvider.Meta().(*clients.Client).Account.Environment.APIManagementHostNameSuffix
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data, b2cConfig, apiDomain),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config:      r.requiresImport(data, b2cConfig, apiDomain),
+			ExpectError: acceptance.RequiresImportError(data.ResourceType),
+		},
+	})
+}
+
+func testAccAzureRMApiManagementIdentityProviderAADB2C_getB2CConfig(t *testing.T) map[string]string {
+	config := map[string]string{
+		"tenant_id":     "",
+		"tenant_slug":   "",
+		"client_id":     "",
+		"client_secret": "",
+	}
+
+	for k := range config {
+		e := fmt.Sprintf("ARM_TEST_B2C_%s", strings.ToUpper(k))
+		if v := os.Getenv(e); v != "" {
+			config[k] = v
+			continue
+		}
+		vars := make([]string, 0, len(config))
+		for k := range config {
+			v := fmt.Sprintf("ARM_TEST_B2C_%s", strings.ToUpper(k))
+			vars = append(vars, v)
+		}
+		t.Skip(fmt.Sprintf("Acceptance tests for resource `azurerm_api_management_identity_provider_aadb2c` skipped unless environment variables set: %s", strings.Join(vars, ", ")))
+	}
+
+	return config
+}
+
+func (ApiManagementIdentityProviderAADB2CResource) Exists(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	id, err := parse.IdentityProviderID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.ApiManagement.IdentityProviderClient.Get(ctx, id.ResourceGroup, id.ServiceName, apimanagement.IdentityProviderType(id.Name))
+	if err != nil {
+		return nil, fmt.Errorf("reading ApiManagement Identity Provider AADB2C (%s): %+v", id, err)
+	}
+
+	return utils.Bool(resp.ID != nil), nil
+}
+
+func (ApiManagementIdentityProviderAADB2CResource) basic(data acceptance.TestData, b2cConfig map[string]string, apiDomain string) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+provider "azuread" {
+  tenant_id     = "%[1]s"
+  client_id     = "%[2]s"
+  client_secret = "%[3]s"
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-api-%[5]d"
+  location = "%[6]s"
+}
+
+resource "azurerm_api_management" "test" {
+  name                = "acctestAM-%[5]d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  publisher_name      = "pub1"
+  publisher_email     = "pub1@email.com"
+  sku_name            = "Developer_1"
+}
+
+resource "azuread_application" "test" {
+  name                       = "acctestAM-%[5]d"
+  oauth2_allow_implicit_flow = true
+  reply_urls                 = ["https://${azurerm_api_management.test.name}.developer.%[8]s/signin"]
```

Comment:
we should probably expose this as a computed field fwiw (the `developer` uri?)

Reply:
Good tip, turns out this is already exported in azurerm_api_management :)